src/backend/distributed/planner/multi_explain.c

```diff
@@ -399,15 +438,31 @@ RemoteExplain(Task *task, ExplainState *es)
 
 	RemoteExplainPlan *remotePlan = (RemoteExplainPlan *) palloc0(
 		sizeof(RemoteExplainPlan));
-	StringInfo explainQuery = BuildRemoteExplainQuery(TaskQueryStringForAllPlacements(
-														  task),
-													  es);
 
-	/*
-	 * Use a coordinated transaction to ensure that we open a transaction block
-	 * such that we can set a savepoint.
-	 */
-	UseCoordinatedTransaction();
```

Comment:
We should always use a coordinated transaction in case there are intermediate pgbouncers. Otherwise, the  UDF calls may go into different sessions than the actual query.